umbraco/Umbraco-CMS
# Removing more membership providers from being used

With this update, membership providers are no longer used for back office operations for members (already have been removed for users). So now membership provider functionality only remains for the front-end which will be easier to manage when we split the web projects into front-end and back end. This also means there's a lot less usages of System.Web.Security.

Eventually membership will need to be replaced with aspnet identity in the front-end which is fine but for now, the back office allows editing member data by talking directly to our services and security/password handling is dealt with by some abstractions.

This also removes all of the old legacy logic of having to deal with the huge amount of legacy membership provider settings and password changing. For example, all of this is removed:

* enablePasswordRetrieval
* requiresQuestionAnswer
* enableReset
* resetPassword
* generatedPassword / GeneratedPassword

And sorts out the simple usage of allowManuallyChangingPassword / AllowManuallyChangingPassword

Changing passwords  in the back office is now simple:

* You can change any user's password apart from your own if you have access to the user section
* You can change your own password if you supply your existing one
* You can change any members password if you have access to edit them

There's no more password resetting/generating via the back office for members. We could bring this back if we want to but that was more of a work around for legacy behavior. For now it's gone though which is fine for the time being.

You'll notice there's a new IMemberPasswordConfiguration and it's implementation is just a static config class just like Users. Eventually when we get aspnet identity in for members, we'll need to replace that implementation.

## Testing

* Make sure you can login
* Change your own password
* Create other users
* Login as them
* Change other users' passwords
* Create members in the back office, assign roles, change passwords
* Create 3x partial view macros for the front end - register member, login member, member status
   * Allow these macros to be rendered in an RTE
   * Put these macros in an RTE then test that you can register, login, logout on the front-end
